lib/util/fs/index.ts

```diff
@@ -11,6 +11,15 @@ export * from './proxies';
 
 export const pipeline = util.promisify(stream.pipeline);
 
+function isPathInBaseDir(path: string, baseDir?: string): boolean {
+  if (baseDir && !path.startsWith(upath.resolve(baseDir))) {
+    logger.warn('Preventing access to file outside the base directory');
```

Comment:
```suggestion
    logger.warn({ path, baseDir }, 'Preventing access to file outside the base directory');
```

May need prettifying